sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -209,4 +277,45 @@ protected void sleepIfRunningAgainstService(long millis) {
             throw logger.logExceptionAsWarning(new IllegalStateException(ex));
         }
     }
+
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    private static int getOffset() {
+        if (testMode == TestMode.PLAYBACK) {
+            return 0;
+        }
+        LocalDate today = LocalDate.now(ZoneId.systemDefault());
+        return (today.getDayOfWeek().getValue() + getPlatFormOffset()) % PLATFORM_COUNT;
+    }
+
+    private static Integer getPlatFormOffset() {
+        String currentOs = System.getProperty("os.name").toLowerCase(Locale.ROOT);
+        System.out.println("It is currently running on os: " + currentOs);
+        String currentJdk = System.getProperty("java.version").toLowerCase(Locale.ROOT);
+        System.out.println("It is using jdk: " + currentJdk);
+
+        for (int i = 0; i < PLATFORM_LIST.size(); i++) {
+            if (currentOs.toLowerCase(Locale.ROOT).contains(PLATFORM_LIST.get(i).split(",")[0].toLowerCase(Locale.ROOT))
+                && currentJdk.toLowerCase(Locale.ROOT).contains(
+                    PLATFORM_LIST.get(i).split(",")[1].toLowerCase(Locale.ROOT))) {
+                return i;
+            }
```

Comment:
This code should be made more flexible, to support other versions of the JDK other than 8 and 11. I would expect every JDK version less than 8 to be an exception, and every version after 8 to be considered 'lts' (i.e. long term support).

Reply:
This piece of code only compare with platformList and get the index of platform in the list. If say, Java 14 is supported in pipeline live tests, to make the code work we need to add it to PLATFORM_LIST instead of checking here. The code here only affects rolling tests in live. 

If we only want to test Java 14 in playback and record other than rolling, then it works without touching the code.

Reply:
It is currently reading from file under eng. This helps avoid re-releasing every time we update the list.